pkg/resources/scim_integration_acceptance_test.go

```diff
@@ -170,11 +176,102 @@ func TestAcc_ScimIntegration_invalid(t *testing.T) {
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"name": config.StringVariable(id.Name()),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		ErrorCheck: helpers.AssertErrorContainsPartsFunc(t, []string{
+			`The argument "enabled" is required, but no definition was found.`,
+			`The argument "scim_client" is required, but no definition was found.`,
+			`The argument "run_as_role" is required, but no definition was found.`,
+		}),
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_ScimIntegration/invalid"),
+				ConfigVariables: m(),
+			},
+		},
+	})
+}
+
+func TestAcc_ScimIntegration_migrateFromVersion091(t *testing.T) {
+	role, roleCleanup := acc.TestClient().Role.CreateRoleWithRequest(t, sdk.NewCreateRoleRequest(snowflakeroles.GenericScimProvisioner).WithOrReplace(true))
+	t.Cleanup(roleCleanup)
+	acc.TestClient().Role.GrantRoleToCurrentRole(t, role.ID())
+	id := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	resourceName := "snowflake_scim_integration.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"snowflake": {
+						VersionConstraint: "=0.91.0",
+						Source:            "Snowflake-Labs/snowflake",
+					},
+				},
+				Config: scimIntegrationv091(id.Name(), role.Name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", id.Name()),
+					resource.TestCheckResourceAttr(resourceName, "provisioner_role", role.ID().Name()),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				Config:                   scimIntegrationv092(id.Name(), role.Name),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{plancheck.ExpectEmptyPlan()},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", id.Name()),
+					resource.TestCheckResourceAttr(resourceName, "run_as_role", role.ID().Name()),
+				),
+			},
+		},
+	})
+}
+
+func scimIntegrationv091(name, roleName string) string {
+	s := `
+resource "snowflake_scim_integration" "test" {
+	name             = "%s"
+	scim_client      = "%s"
+	provisioner_role = "%s"
+}
+`
+	return fmt.Sprintf(s, name, sdk.ScimSecurityIntegrationScimClientGeneric, roleName)
+}
+
+func scimIntegrationv092(name, roleName string) string {
+	s := `
+resource "snowflake_scim_integration" "test" {
+	name             = "%s"
+	enabled          = true
```

Comment:
how is this empty plan if the enabled was changed to true? In other comment you said that it was disabled by default so we should see a change. So:
1. Please change this test to enabled false.
2. Add a separate migration one, where enabled is set to true and we expect a change.

Reply:
Sorry, I got confused with defaults for `ENABLED` because `property_default` is `false`, but when you run a create without `ENABLED` (it's actually possible, contrary to the docs), the `property_value` is `true`.